src/com/senlainc/warsaw/tyurin/TankProduct.java

```diff
@@ -0,0 +1,22 @@
+package com.senlainc.warsaw.tyurin;
+
+public class TankProduct implements IProduct{
+    private Body body;
```

Comment:
If you change it on abstraction (interfaces) class-cast will not be needed lower in code. better always useabstraction